sympy/combinatorics/homomorphisms.py

```diff
@@ -199,29 +244,42 @@ def homomorphism(domain, codomain, gens, images=[]):
     gens.extend([g for g in generators if g not in gens])
     images = dict(zip(gens,images))
 
-    if not _check_homomorphism(domain, images, codomain.identity):
+    if check and not _check_homomorphism(domain, codomain, images):
         raise ValueError("The given images do not define a homomorphism")
     return GroupHomomorphism(domain, codomain, images)
 
-def _check_homomorphism(domain, images, identity):
+def _check_homomorphism(domain, codomain, images):
     rels = domain.relators
+    identity = codomain.identity
     def _image(r):
         if r.is_identity:
             return identity
         else:
             w = identity
             r_arr = r.array_form
             i = 0
+            j = 0
             while i < len(r):
```

Comment:
That would be more convenient for loops but I'm not sure if it is possible. At the moment, a group element is basically defined as an array of tuples like this (with appropriate operations for combining them). If `gen` is made to be a `FreeGroupElement` itself, then group elements might end up being defined in terms of themselves and that sounds circular so could cause problems. Or maybe it'll be fine. I'll look into it. Even if it isn't possible, it could be convenient to have another attribute like that.